### PR TITLE
r3d: add pic variant

### PR DESCRIPTION
**Pull request by rbberger**
This adds the necessary `-fPIC` for linking to this library in some cases.

### CODE REVIEW

File: var/spack/repos/builtin/packages/r3d/package.py [hunk 1]
```diff
@@ -36,6 +36,8 @@ class R3d(CMakePackage):
         description="Build R3D regression tests (versions 2019-04-24 or earlier)",
     )
 
+    variant("shared", default=False, description="Build position independent code")
```
Comment (raovgarimella) [inline var/spack/repos/builtin/packages/r3d/package.py]: Does spack not have a built-in shared variant?

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/r3d/package.py]: The only variant defined in (so inherited from) `CMakePackage` is `generator`.

Comment (tldahlgren) [inline var/spack/repos/builtin/packages/r3d/package.py]: Nearly all packages that want to set the option below use `pic` for this variant (e.g., https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/netcdf-c/package.py).

**Review by tldahlgren — COMMENTED** [review]
I suggest using a more "standard" variant for the option; otherwise, LGTM.

File: var/spack/repos/builtin/packages/r3d/package.py (hunk 1 above)
Comment (rbberger) [reply var/spack/repos/builtin/packages/r3d/package.py]: @tldahlgren done.

**Review by tldahlgren — APPROVED** [review]
LGTM.

**Comment by tldahlgren** [comment]
Will defer merge to give @raovgarimella and @gaber a change to comment post-change.

**Review by gaber — APPROVED** [review]
LGTM!

**Comment by raovgarimella** [comment]
<Thumbs up>

**Comment by tldahlgren** [comment]
> No description provided.

@raovgarimella Is that an approval?

**Comment by tldahlgren** [comment]
Oops.  Meant to add comment.

**Comment by raovgarimella** [comment]
Yes. Approved.

**Pull request merged**